src/hasher.rs

```diff
@@ -1,28 +1,83 @@
 use core::default::Default;
 use core::fmt;
 use core::fmt::Write;
+use core::ops::DerefMut;
 use arrayvec::{ArrayVec};
 use nanos_sdk::bindings::*;
 use zeroize::{Zeroize, Zeroizing};
-use base64;
 
 pub trait Hasher<const N: usize> {
```

Comment:
Does this work?
```suggestion
pub trait Hasher {
    const N: usize;
```
to show that each type will have a unique `Hasher` instance with one possible `N`?